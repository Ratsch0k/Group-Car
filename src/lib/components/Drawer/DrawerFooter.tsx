import React from 'react';
import {Grid, Link} from '@material-ui/core';
import useModalRouter from 'lib/hooks/useModalRouter';
import {useTranslation} from 'react-i18next';

type GridProps = import('@material-ui/core').GridProps;

interface DrawerFooterProps extends GridProps {
  justify?: undefined;
}

const DrawerFooter: React.FC<DrawerFooterProps> =
(props: DrawerFooterProps) => {
  const {...rest} = props;
  const {goTo} = useModalRouter();
  const {t} = useTranslation();

  return (
    <Grid container {...rest} justify='space-evenly'>
      <Grid item>
        <Link component='button' onClick={() => goTo('/privacy-policy')}>
          {t('privacyPolicy.title')}
        </Link>
      </Grid>
      <Grid item>
        <Link component='button' onClick={() => goTo('/imprint')}>
          {t('imprint.title')}
        </Link>
      </Grid>
    </Grid>
  );
};

export default DrawerFooter;
