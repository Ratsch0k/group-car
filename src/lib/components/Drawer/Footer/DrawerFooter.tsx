import React from 'react';
import {Grid, Link, Container} from '@material-ui/core';
import {
  GroupCarTheme,
} from 'lib';
import {useTranslation} from 'react-i18next';
import {makeStyles, createStyles} from '@material-ui/styles';
import {useAppDispatch} from 'lib/redux/hooks';
import {goToModal} from 'lib/redux/slices/modalRouter/modalRouterSlice';


/**
 * Element.
 */
export const DrawerFooter: React.FC = () => {
  const {t} = useTranslation();
  const dispatch = useAppDispatch();

  /**
 * Styles
 */
  const useStyles = makeStyles((theme: GroupCarTheme) =>
    createStyles({
      paper: {
        width: '100%',
      },
      container: {
        padding: theme.spacing(1),
      },
    }),
  );
  const classes = useStyles();

  return (
    <Container className={classes.container}>
      <Grid container justifyContent='space-evenly'>
        <Grid item>
          <Link component='button' onClick={
            () => dispatch(goToModal('/privacy-policy'))
          }>
            {t('privacyPolicy.title')}
          </Link>
        </Grid>
        <Grid item>
          <Link
            component='button'
            onClick={() => dispatch(goToModal('/imprint'))}
          >
            {t('imprint.title')}
          </Link>
        </Grid>
      </Grid>
    </Container>
  );
};

export default DrawerFooter;
