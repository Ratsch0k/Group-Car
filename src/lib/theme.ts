import {createMuiTheme} from '@material-ui/core';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#00C68A',
      light: '#98F0D6',
      dark: '#00A473',
      contrastText: '#FFFFFF',
    },
    secondary: {
      main: '#AD00D8',
      light: '#ECA1FF',
      dark: '#8900AB',
      contrastText: '#FFFFFF',
    },
  },
});

export default theme;
