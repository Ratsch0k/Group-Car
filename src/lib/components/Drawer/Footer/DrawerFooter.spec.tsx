import React from 'react';
import {render} from '@testing-library/react';
import DrawerFooter from './DrawerFooter';
import { ThemeProvider } from '@material-ui/core';
import theme from '../../../../__test__/testTheme';

it('renders without crashing', () => {
  render(
    <ThemeProvider theme={theme}>
      <DrawerFooter />
    </ThemeProvider>);
});

it('matches snapshot', () => {
  const {container} = render(
    <ThemeProvider theme={theme}>
      <DrawerFooter />
    </ThemeProvider>
  );

  expect(container).toMatchSnapshot();
});
