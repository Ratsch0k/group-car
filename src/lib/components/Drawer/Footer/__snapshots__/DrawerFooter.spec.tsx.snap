// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root makeStyles-container MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-justify-content-xs-space-evenly"
    >
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
        >
          privacyPolicy.title
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
      >
        <button
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiLink-button MuiTypography-colorPrimary"
        >
          imprint.title
        </button>
      </div>
    </div>
  </div>
</div>
`;
